frontend/src/components/HierarchySwitcher.jsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react';
+import { Button, Menu, MenuItem } from '@mui/material';
+import { useNavigation } from './NavigationContext';
+import { fetchHierarchies } from '../api'; // Make sure this import points to your API fetching logic
+
+function HierarchySwitcher() {
+  const { selectedHierarchy, setSelectedHierarchy, setSelectedRegion } = useNavigation();
+  const [hierarchies, setHierarchies] = useState([]);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+
+  useEffect(() => {
+    const getHierarchies = async () => {
+      try {
+        const fetchedHierarchies = await fetchHierarchies();
+        setHierarchies(fetchedHierarchies);
+      } catch (error) {
+        console.error('Error fetching hierarchies:', error);
+      }
+    };
+
+    getHierarchies();
+  }, []);
+
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleHierarchyChange = (hierarchyId) => {
+    setSelectedHierarchy({ hierarchyId });
+    // Reset the selected region to the world
+    setSelectedRegion({
+      id: null,
+      name: 'World',
+      info: {},
+      hasSubregions: false,
+    });
```

Comment:
**Suggestion:** Consider using a constant for the 'World' region object as it is used multiple times in the code.
```suggestion
  const WORLD_REGION = { id: null, name: 'World', info: {}, hasSubregions: false };

  const handleHierarchyChange = (hierarchyId) => {
    setSelectedHierarchy({ hierarchyId });
    // Reset the selected region to the world
    setSelectedRegion(WORLD_REGION);
    handleClose();
  };
```